validation/views.py

```diff
@@ -372,6 +485,102 @@ def register(request):
 # TODO: Speaker bio page like https://ojibwe.lib.umn.edu/about/voices
 
 
+@login_required()
+@require_http_methods(["POST"])
+def record_translation_judgement(request, phrase_id):
+    phrase = get_object_or_404(Phrase, id=phrase_id)
+    judgement = json.loads(request.body)
+
+    if judgement["judgement"] == "yes":
+        phrase.validated = True
+        phrase.status = "linked"
+        phrase.modifier = str(request.user)
+        phrase.date = datetime.datetime.now()
+    elif judgement["judgement"] in ["no", "idk"]:
+        phrase.validated = False
+        phrase.status = "new"
+        phrase.modifier = str(request.user)
+        phrase.date = datetime.datetime.now()
+    else:
+        return HttpResponseBadRequest()
+
+    phrase.save()
+    return JsonResponse({"status": "ok"})
+
+
+@login_required()
+@require_http_methods(["POST"])
+def record_audio_quality_judgement(request, recording_id):
+    rec = get_object_or_404(Recording, id=recording_id)
+    judgement = json.loads(request.body)
+
+    if judgement["judgement"] in ["good", "bad"]:
+        rec.quality = judgement["judgement"]
+    else:
+        return HttpResponseBadRequest()
+
+    rec.save()
+    return JsonResponse({"status": "ok"})
+
+
 def encode_query_with_page(query, page):
     query["page"] = page
     return f"?{query.urlencode()}"
+
+
+def user_is_linguist(user):
+    if user.is_authenticated:
+        for g in user.groups.all():
+            if g.name == "Linguist":
+                return True
```

Comment:
This whole method can be replaced with `user.groups.filter(name='Linguist').exists()`, and the next one with `user.groups.filter(name__in=['Linguist', 'Expert']).exists()`. If the user is not authenticated, `user` is an [`AnonymousUser`](https://docs.djangoproject.com/en/3.1/ref/contrib/auth/#anonymoususer-object) object on which `user.groups` will always return an empty queryset, and `False` for `.exists()`.

Reply:
Nice!